react/components/HighlightOverlay/index.tsx

```diff
@@ -81,35 +85,44 @@ const HighlightOverlay: React.FC<Props> = props => {
 
   return (
     <>
-      <div
-        id="editor-provider-overlay"
-        style={highlightStyle}
-        className={`absolute bw2 ba ${
-          (hasValidElement && hasHighlight) || openBlockTreePath
-            ? 'o-100'
-            : 'o-0'
-        }`}
+      <CSSTransition
+        in={Boolean((hasValidElement && hasHighlight) || openBlockTreePath)}
+        classNames={{
+          enter: styles['highlight-enter'],
+          enterActive: styles['highlight-enter-active'],
+          enterDone: styles['highlight-enter-done'],
+          exit: styles['highlight-exit'],
+          exitActive: styles['highlight-exit-active'],
+          exitDone: styles['highlight-exit-done'],
+        }}
+        mountOnEnter
+        timeout={150}
       >
-        {title && (
-          <p
-            className="absolute c-action-secondary f7 ma0 right-0 ph2 pb2 pt1 truncate tc"
-            style={labelStyle}
-            title={title}
-          >
-            {title}
-          </p>
-        )}
-      </div>
-      {hasValidElement &&
-        openBlockTreePath &&
-        portalContainer &&
-        ReactDOM.createPortal(
-          <div
-            className="absolute bg-base h-100 w-100 o-40"
-            style={maskStyle}
-          />,
-          portalContainer
-        )}
+        <div
+          id="editor-provider-overlay"
+          style={highlightStyle}
+          className="absolute bw2 ba"
+        >
+          {title && (
+            <p
+              className="absolute c-action-secondary f7 ma0 right-0 ph2 pb2 pt1 truncate tc"
+              style={labelStyle}
+              title={title}
```

Comment:
What is `title` for?

Reply:
I added the `title` for the case in which the text overflows and since I added `truncate`, to user can hover the mouse to show the full text, like so:
[Codepen](https://codepen.io/thiagomurakami/pen/NWWRxEj)
<img width="1049" alt="Screen Shot 2019-10-17 at 10 18 40" src="https://user-images.githubusercontent.com/10400340/67012351-a04bef80-f0c7-11e9-8b89-e8bdc523f59c.png">

Reply:
Cool! Didn't know that!

Reply:
However it isn't working because of `pointer-events: none` 😞 So I guess I can remove it...